README.md

```diff
@@ -1,25 +1,51 @@
-# in-toto-kubesec
+# in-toto-kubectl
 
 This is a kubectl plugin to run in-toto verification on the images in your
 kubernetes pods.
 
-### Install
+## Install
 
 run `make deploy` and the plugin should be installed to `~/.kube/plugins`. You
 can change the target by changing the KUBEPATH environment variable. For
 example `make deploy KUBEPATH=~/bin` will install it to a user-controlled
 `bin/` folder.
 
-### Usage
+## Usage
 
 Make sure the plugin executable was installed to somewhere in your `$PATH`, or
 to add `~/.kube/plugins` to your path. Afterwards, you can use it within
 kubectl:
 
 ```bash
-kubectl plugin in-toto pod/[podname]
+kubectl in-toto pod/[podname]
 ```
 
-### Credit
+In order to scan a pod, you'd have to have the link metadata and the layout in
+your current folder. After passing the pod/podname argument, you can also use
+`-k` and `-l` in the same way as `in-toto-verify` to pass key and layout
+parameters.
+
+### Extensions
+
+The kubectl plugin uses parameter substitution to provide you with a
+`{IMAGE_ID}` parameter that you can substitute inside of your layouts. 
+
+In addition, a file (if it doesn't exist) called `image_id` will be populated
+on the directory when verification starts. This can be used to e.g., verify
+against the output of `docker build`. This second extension will disappear in
+future releases, and once resource type identifiers are provided by the in-toto
+framework.
+
+
+## Example
+
+An example repository exists under the `example` directory. It contains all the
+tools you need to create a layout (using the python implementation), create
+signed metadata files (you will need docker to build the container). If you're
+using minikube to run the example, I also suggest you expose the Docker socket
+before executing the functionary step so as to create the image inside the
+container.
```

Comment:
Here's a single copy-pastable snippet to run through the example.
```bash
## Install kubectl plugin
# Also see https://kubernetes.io/docs/tasks/tools/install-kubectl/
make deploy
export PATH=~/.kube/plugins/:$PATH


# Start minikube
# Also see https://kubernetes.io/docs/tasks/tools/install-minikube/#install-minikube
minikube start


## Do the demo functionary work
cd example/functionary/

# NOTE: Compile on a system compatible with your `FROM scratch` container
gcc --static main.c

# Prepare for building docker image inside minikube
# Also see https://kubernetes.io/docs/setup/minikube/#use-local-images-by-re-using-the-docker-daemon
eval $(minikube docker-env)

# Build docker image generating signed in-toto link metadata
# Also see https://github.com/in-toto/in-toto/#installation
in-toto-run -k functionary -n build -p image_id -- docker build . -t empty:1.0.0 --iidfile image_id


## Spin up container image inside kubernetes cluster
# NOTE: Use `kubectl get pods` to see if the pods running
kubectl apply -f ../deployments/test.yml


## In-toto verification
cd ../verification/
# Cache pod name (required below)
pod_name=$(kubectl get pods --no-headers --output custom-columns=":metadata.name")

# Fetch in-toto link metadata from build functionary
# NOTE: If `root.layout` isn't there or has expired re-create using
# `../project-owner/create_layout.py` and copy it over too.
# `root_key.pub` is also required in this directory.
cp ../functionary/build.f5c8b97a.link .

# Verify that the software supply chain that resulted in the running pod was
# carried out according to its definition in a validly signed root.layout
kubectl in-toto -k root_key.pub -l root.layout pod/${pod_name}
```

Please let me know if I got it right.